nethermind/quickstart.sh

```diff
@@ -0,0 +1,652 @@
+
+    # Check RAM memory space
+    check_ram_memory_space
+
+    # Check node arguments
+    check_node_arguments
+
+    echo -e "\nSanity checks finished successfully!"
+}
+
+# Setup - prepare node to run Fuse client
+function setup() {
+    echo -e "\nSetup..."
+
+    # Install and configure NTP
+    install_ntp
+
+    # Specify image versions
+    FUSE_CLIENT_DOCKER_REPOSITORY="fusenet/nethermind-node"
+    FUSE_VALIDATOR_DOCKER_REPOSITORY="fusenet/validator-app"
+    SPARK_VALIDATOR_DOCKER_REPOSITORY="fusenet/spark-validator-app"
+
+    NETSTATS_CLIENT_DOCKER_REPOSITORY="fusenet/netstat"
+    SPARK_NETSTATS_CLIENT_DOCKER_REPOSITORY="fusenet/spark-netstat"
+
+    FUSE_CLIENT_DOCKER_IMAGE_VERSION="1.13.3"
+    FUSE_VALIDATOR_DOCKER_IMAGE_VERSION="1.0.0"
+    SPARK_VALIDATOR_DOCKER_IMAGE_VERSION="1.0.0"
+
+    NETSTATS_CLIENT_DOCKER_IMAGE_VERSION="1.0.0"
+    SPARK_NETSTATS_CLIENT_DOCKER_IMAGE_VERSION="1.0.0"
+
+    FUSE_CLIENT_DOCKER_IMAGE=$FUSE_CLIENT_DOCKER_REPOSITORY:$FUSE_CLIENT_DOCKER_IMAGE_VERSION
+    FUSE_VALIDATOR_DOCKER_IMAGE=$FUSE_VALIDATOR_DOCKER_REPOSITORY:$FUSE_VALIDATOR_DOCKER_IMAGE_VERSION
+    SPARK_VALIDATOR_DOCKER_IMAGE=$SPARK_VALIDATOR_DOCKER_REPOSITORY:$SPARK_VALIDATOR_DOCKER_IMAGE_VERSION
+    NETSTATS_CLIENT_DOCKER_IMAGE=$NETSTATS_CLIENT_DOCKER_REPOSITORY:$NETSTATS_CLIENT_DOCKER_IMAGE_VERSION
+    SPARK_NETSTATS_CLIENT_DOCKER_IMAGE=$SPARK_NETSTATS_CLIENT_DOCKER_REPOSITORY:$SPARK_NETSTATS_CLIENT_DOCKER_IMAGE_VERSION
+
+    echo -e "\nFuse - Client: $FUSE_CLIENT_DOCKER_IMAGE_VERSION"
+
+    if [[ $NETWORK == "spark" ]]; then
+        # Print versions
+        echo -e "Fuse - Netstat: $SPARK_NETSTATS_CLIENT_DOCKER_IMAGE_VERSION"
+
+        echo -e "\nPull Docker images...\n"
+
+        # Pull needed Docker images
+        $PERMISSION_PREFIX docker pull $FUSE_CLIENT_DOCKER_IMAGE
+        $PERMISSION_PREFIX docker pull $SPARK_NETSTATS_CLIENT_DOCKER_IMAGE
+
+        if [[ $ROLE == "validator" ]]; then
+            $PERMISSION_PREFIX docker pull $SPARK_VALIDATOR_DOCKER_IMAGE
+        fi
+    else
+        # Print versions
+        echo -e "\nFuse - Client: $FUSE_CLIENT_DOCKER_IMAGE_VERSION"
+        echo -e "Fuse - Netstat: $NETSTATS_CLIENT_DOCKER_IMAGE_VERSION"
+
+        echo -e "\nPull Docker images...\n"
+
+        # Pull needed Docker images
+        $PERMISSION_PREFIX docker pull $FUSE_CLIENT_DOCKER_IMAGE
+        $PERMISSION_PREFIX docker pull $NETSTATS_CLIENT_DOCKER_IMAGE
+
+        if [[ $ROLE == "validator" ]]; then
+            $PERMISSION_PREFIX docker pull $FUSE_VALIDATOR_DOCKER_IMAGE
+        fi
+    fi
+
+    # Directories
+    BASE_DIR="$(pwd)/fusenet"
+    DATABASE_DIR=$BASE_DIR/database
+    LOGS_DIR=$BASE_DIR/logs
+    KEYSTORE_DIR=$BASE_DIR/keystore
+
+    # Create needed directories
+    mkdir -p $BASE_DIR
+    mkdir -p $DATABASE_DIR
+    mkdir -p $LOGS_DIR
+    mkdir -p $KEYSTORE_DIR
+
+    # Generate keystore file
+    if [[ $ROLE == "validator" ]]; then
+        generate_eth_private_key
```

Comment:
Only want to run this is keystore directory is empty (otherwise you overwrite the keys each run)